reactive/kotlinx-coroutines-rx2/test/SchedulerTest.kt

```diff
@@ -31,4 +38,252 @@ class SchedulerTest : TestBase() {
         }
         finish(4)
     }
+
+    @Test
+    fun testAsSchedulerWithNoDelay(): Unit = runTest {
+        expect(1)
+        val mainThread = Thread.currentThread()
+        val scheduler = (currentDispatcher() as CoroutineDispatcher).asScheduler()
+        suspendCancellableCoroutine<Unit> {
+            scheduler.scheduleDirect {
+                val t1 = Thread.currentThread()
+                assertSame(t1, mainThread)
+                expect(2)
+                it.resume(Unit)
+            }
+        }
+        yield()
+        finish(3)
+    }
+
+    @Test
+    fun testAsSchedulerWithDelay(): Unit = runTest {
+        expect(1)
+        val mainThread = Thread.currentThread()
+        val scheduler = (currentDispatcher() as CoroutineDispatcher).asScheduler()
+        val delayMillis = 300L
```

Comment:
This is going to slow down test runs. I think you could split this test into two: one with a minimal delay (1ms) that verifies the dispatcher is actually being used (like in the non-delay case), and one that uses a `TestCoroutineDispatcher` to fast-forward through a longer delay to verify that the delay is actually being respected.